[rocsparse] Add Disclaimers regarding hipSPARSE (#943)
ROCm/rocm-libraries: projects/rocsparse/README.md

# rocSPARSE

> **⚠️ Important: hipSPARSE is the recommended library for development**
> 

> > For new development projects, we strongly recommend using **[hipSPARSE](https://github.com/ROCm/rocm-libraries/tree/develop/projects/hipsparse)** instead of rocSPARSE. 

Reviewer: ```suggestion
> For new development projects, **[hipSPARSE](https://github.com/ROCm/rocm-libraries/tree/develop/projects/hipsparse)** is recommended instead of rocSPARSE. 
```

> hipSPARSE provides a comprehensive, portable interface that supports multiple backends (including rocSPARSE and cuSPARSE) and is the preferred library for sparse matrix operations.
> 

> > For documentation and examples, please refer to the [hipSPARSE documentation](https://rocm.docs.amd.com/projects/hipSPARSE/en/latest/).

Reviewer: ```suggestion
> For documentation and examples, see the [hipSPARSE documentation](https://rocm.docs.amd.com/projects/hipSPARSE/en/latest/).
```

> 

> > This rocSPARSE repository is maintained for reference purposes and to document any specialized functionality that may not be available in hipSPARSE.

Reviewer: ```suggestion
> This rocSPARSE repository is maintained for reference purposes and to document any specialized functionality that might not be available in hipSPARSE.
```
